include/RAJA/policy/hip/kernel/internal.hpp

```diff
@@ -410,6 +410,26 @@ using hip_statement_list_executor_t = HipStatementListExecutor<
     StmtList,
     Types>;
 
+struct HIPKernelLaunchFuncData {
+  static unsigned int getKernelScratchBytes(
+      hipFuncAttributes const &hip_func_attributes) {
+    return hip_func_attributes.localSizeBytes;
+  }
+  static unsigned int getKernelRegisters(
+  	hipFuncAttributes const &hip_func_attributes) {
+	 return hip_func_attributes.numRegs;
+   }
+
+
+static hipFuncAttributes get_hip_func_attributes(void const *kernel_func) {
+    static hipFuncAttributes attr = [=]() {
+      hipFuncAttributes attr;
+      hipFuncGetAttributes(&attr, kernel_func);
```

Comment:
Should this be error checked?

Reply:
I was looking for a HIP_SAFE_CALL macro defined somewhere. Do we have one?

Reply:
Take a look here https://github.com/LLNL/RAJA/blob/develop/include/RAJA/policy/hip/raja_hiperrchk.hpp#L45